### ISSUE TITLE
Reduce unncessary nodes in parse tree.

### ISSUE THREAD

**Issue by ChristopherChudzicki** [mitodl/mitx-grading-library#60]
The parser in `calc.py` attaches `sum/product/parallel/power` to *every* set of parentheses, even if there's nothing to add/multiply/etc. For example, `((2))` parses as:

```xml
<sum>
  <product>
    <parallel>
      <negation>
        <power>
          <atom>
            <parentheses>
              <sum>
                <product>
                  <parallel>
                    <negation>
                      <power>
                        <atom>
                          <parentheses>
                            <sum>
                              <product>
                                <parallel>
                                  <negation>
                                    <power>
                                      <atom>
                                        <number>
                                          <ITEM>2</ITEM>
                                        </number>
                                      </atom>
                                    </power>
                                  </negation>
                                </parallel>
                              </product>
                            </sum>
                          </parentheses>
                        </atom>
                      </power>
                    </negation>
                  </parallel>
                </product>
              </sum>
            </parentheses>
          </atom>
        </power>
      </negation>
    </parallel>
  </product>
</sum>
```

it would be nicer (more readable during debugging) we could through away grouped nodes with only 1 child.

**Comment by jolyonb** [comment]
I would love this. I couldn't figure out how to get pyparsing to do it during my refactoring though, so I went for the method I knew would work. I think this would also significantly speed up both the parsing and evaluation time.

**Comment by jolyonb** [comment]
Addressed in #62.
